tests/subscriptions/test_worker.py

```diff
@@ -121,16 +127,29 @@ def test_subscription_worker(dataset: Dataset, time_shift: Optional[timedelta])
         message = consumer.poll()
         assert message is not None
         assert message.partition.topic == result_topic
-        assert message.payload.task == results[i].task
-        assert message.payload.task.timestamp == timestamp
-        assert message.payload.result == results[i].future.result()
-        request, result = message.payload.result
-        assert request.extensions["timeseries"] == {
-            "from_date": (timestamp - subscription.data.time_window).isoformat(),
-            "to_date": timestamp.isoformat(),
-            "granularity": 3600,  # XXX: unused, no time grouping
-        }
-        assert result == {
+
+        task_result_future = result_futures[i]
+        expected_payload = subscription_task_result_codec.encode(
+            SubscriptionTaskResult(
+                task_result_future.task, task_result_future.future.result()
+            )
+        )
+
+        assert message.payload == expected_payload
+
+        decoded_payload = json.loads(expected_payload.value)["payload"]
+        assert decoded_payload["timestamp"] == timestamp.isoformat()
+        assert (
+            # NOTE: The time series extension is folded back into the request
+            # body, ideally this would reference the timeseries options in
+            # isolation.
+            decoded_payload["request"].items()
+            > {
+                "from_date": (timestamp - subscription.data.time_window).isoformat(),
+                "to_date": timestamp.isoformat(),
+            }.items()
+        )
+        assert decoded_payload["result"] == {
             "meta": [{"name": "count", "type": "UInt64"}],
             "data": [{"count": 0}],
         }
```

Comment:
This test is kind of clunky now because the task result codec doesn't work bidirectionally, but it works.